### ISSUE TITLE
Flutter Firebase codelab broken with Cocoapods 1.3.1

### ISSUE THREAD

**Issue by mravn-google** [flutter/flutter#12818]
## Steps to Reproduce

(Happened to more than one user at the Firebase Dev Summit codelab)

On a MacBook with version 1.3.1 of cocoapods installed. [Note: This is the latest stable version according to homebrew at the time of this writing. Confusingly, the `pod` executable, and thereby `flutter doctor`, reports version 1.2.1.

```
$ brew info cocoapods
cocoapods: stable 1.3.1 (bottled)
Dependency manager for Cocoa projects
https://cocoapods.org/
/usr/local/Cellar/cocoapods/1.2.0 (7,567 files, 11.7MB)
  Poured from bottle on 2017-03-09 at 22:52:52
/usr/local/Cellar/cocoapods/1.2.1 (7,646 files, 11.8MB)
  Poured from bottle on 2017-06-30 at 09:47:57
/usr/local/Cellar/cocoapods/1.3.1 (7,746 files, 12.0MB) *
  Poured from bottle on 2017-08-14 at 12:48:13
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/cocoapods.rb
$ pod --version
1.2.1
```
End of note.]

Download the Flutter project at `https://github.com/flutter/friendlychat-steps/blob/master/full_steps/step_0_offline` to get the sample code for the Flutter Firebase code lab (step 3).

Add the dependency
```
google_sign_in: 0.3.1
```
to `pubspec.yaml` (this is part of step 5).

Execute `flutter run` targeting the iOS Simulator. Cocoapods then fails, and we get the following error message:
```
$ flutter run
Running "flutter packages get" in step_0_offline...   0.6s
Launching lib/main.dart on iPhone 7 in debug mode...
Running pod install...                                0.7s
CocoaPods' output:
↳
      Preparing
    
    Analyzing dependencies
    
    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (`arm64`)
    
    Fetching external sources
    -> Fetching podspec for `Flutter` from `/Users/mravn/github/flutter/bin/cache/artifacts/engine/ios`
    -> Fetching podspec for `google_sign_in` from `/Users/mravn/.pub-cache/hosted/pub.dartlang.org/google_sign_in-0.3.1/ios`
    
    Resolving dependencies of `Podfile`
    
    Comparing resolved specification to the sandbox manifest
      A Flutter
      A GTMOAuth2
      A GTMSessionFetcher
      A GoogleSignIn
      A GoogleToolboxForMac
      A google_sign_in
    
    Downloading dependencies
    
    -> Installing Flutter (1.0.0)
    
    -> Installing GTMOAuth2 (1.1.4)
      > Copying GTMOAuth2 from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GTMOAuth2/1.1.4-7fc9c` to `Pods/GTMOAuth2`
    
    -> Installing GTMSessionFetcher (1.1.11)
      > Copying GTMSessionFetcher from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GTMSessionFetcher/1.1.11-5ad62` to `Pods/GTMSessionFetcher`
    
    -> Installing GoogleSignIn (4.1.0)
      > Copying GoogleSignIn from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GoogleSignIn/4.1.0-8c599` to `Pods/GoogleSignIn`
    
    -> Installing GoogleToolboxForMac (2.1.1)
      > Copying GoogleToolboxForMac from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GoogleToolboxForMac/2.1.1-8e329` to `Pods/GoogleToolboxForMac`
    
    -> Installing google_sign_in (0.0.1)
      - Running pre install hooks
    [!] The 'Pods-Runner' target has transitive dependencies that include static binaries: (/Users/mravn/apps/friendlychat-steps/full_steps/step_0_offline/ios/Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework)
    
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:434:in `block (2 levels) in verify_no_static_framework_transitive_dependencies'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:426:in `each'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:426:in `block in verify_no_static_framework_transitive_dependencies'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:423:in `each'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:423:in `verify_no_static_framework_transitive_dependencies'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:113:in `install!'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/command/install.rb:41:in `run'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/command.rb:52:in `run'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/bin/pod:55:in `<top (required)>'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/pod:23:in `load'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/pod:23:in `<main>'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/ruby_executable_hooks:15:in `eval'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/ruby_executable_hooks:15:in `<main>'
Error output from CocoaPods:
↳
    
    [!] `<PBXGroup UUID=`97C146E51CF9000F007C117D`>` attempted to initialize an object with an unknown UUID. `CF3B75C9A7D2FA2A4C99F110` for attribute: `children`. This can be the result of a merge and  the unknown UUID is being discarded.
    
    [!] Automatically assigning platform ios with version 8.0 on target Runner because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
Error running pod install
Error launching application on iPhone 7.
```
Removing the line `use_frameworks!` from `ios/Podfile` fixes the problem.

## Flutter Doctor

```
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-US, channel alpha)
    • Flutter at /Users/mravn/github/flutter
    • Framework revision e8aa40eddd (2 weeks ago), 2017-10-17 15:42:40 -0700
    • Engine revision 7c4142808c
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.0)
    • Android SDK at /Users/mravn/Library/Android/sdk
    • Platform android-27, build-tools 27.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.0.1, Build version 9A1004
    • ios-deploy 1.9.2
    • CocoaPods version 1.2.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2017.2.5)
    • Flutter plugin version 18.3
    • Dart plugin version 172.4343.25

[✓] Connected devices
    • Nexus 5X • 00bff34682e4b2ab                     • android-arm • Android 8.0.0 (API 26)
    • iPhone 7 • 095455A8-E1D3-45DD-B4A4-52A9806655A7 • ios         • iOS 11.0 (simulator)
```

**Comment by mit-mit** [comment]
cc @collinjackson

**Comment by albrecht87** [comment]
Had the same issue on a new project: https://github.com/flutter/flutter/issues/10968#issuecomment-340819651
base issue: https://github.com/flutter/flutter/pull/10205

**Comment by LarkAscending** [comment]
Looks like an inconsistency between versions of the sample code.  We commented out the `use_frameworks!` line in most of the full_steps, but 0 and 1 are lacking the comment character. And all of the offline_steps need this line commented.

**Comment by LarkAscending** [comment]
Also verified this is not an issue with the real cocoapods 1.2.1.

**Comment by mit-mit** [comment]
@LarkAscending can we close this bug? Looks like the issue was resolved?

**Comment by LarkAscending** [comment]
Closed via https://github.com/flutter/friendlychat-steps/pull/52

**Comment by IljaDaderko** [comment]
@LarkAscending I just followed codelab for this, and got errors at `pod install` step. I checked out flutter/friendlychat-steps#52 and then noticed that in my `ios/Podfile` this line is still uncommented. All of these files were generated via intelij plugin. I commented it out and everything got through just fine, but thought it would be good to mention that this is still an issue.

Project was generated today (after mentioned fix was already merged) and is on alpha branch
